src/fstab-generator/fstab-generator.c

```diff
@@ -1459,20 +1459,29 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         } else if (streq(key, "root")) {
 
+                if (generator_soft_rebooted())
```

Comment:
IIRC, root= and friends are used only used in initrd.
But, after soft-reboot system will not enter initrd, right??
In what kind of situation is this necessary??

At least, I think it is better to move the check to add_sysroot_mount() and add_sysroot_usr_mount_fallback().